esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -2691,11 +2691,23 @@ bool CFileSprayEx::onCopy(IEspContext &context, IEspCopy &req, IEspCopyResponse
         StringBuffer destFolder, destTitle, defaultFolder, defaultReplicateFolder;
         StringBuffer srcNodeGroup, destNodeGroup;
         bool bRoxie = false;
+        const char* srcDali = req.getSourceDali();
         const char* destNodeGroupReq = req.getDestGroup();
-        if(!destNodeGroupReq || !*destNodeGroupReq)
+        if(isEmptyString(destNodeGroupReq))
         {
-            getNodeGroupFromLFN(context, srcname, destNodeGroup);
-            DBGLOG("Destination node group not specified, using source node group %s", destNodeGroup.str());
+            CDfsLogicalFileName lfn;
+            lfn.set(srcname);
+            if(!isEmptyString(srcDali) || lfn.isForeign() || lfn.isRemote())
+            {
+                //makes no sense to get the srcname's current group, if a logical file is from a different environment.
+                getDefaultStoragePlane(destNodeGroup);
```

Comment:
>I also noticed that the getDefaultStoragePlane() does not return a plane for bare metal.

In that case, yes, there is only the concept of a default plane in containerized...
This should not call it in bare-metal.
But that means, in bare-metal if no destNodeGroup is supplied - it should fire an error - there is no sensible default.
We could consider allowing a default to be defined in bare-metal to be more consistent, for such times, but I'm not sure it's worth it.